src/openvic-simulation/economy/GoodInstance.cpp

```diff
@@ -3,8 +3,38 @@
 using namespace OpenVic;
 
 GoodInstance::GoodInstance(GoodDefinition const& new_good_definition)
-  : HasIdentifierAndColour { new_good_definition }, good_definition { new_good_definition },
-	price { new_good_definition.get_base_price() }, is_available { new_good_definition.get_is_available_from_start() } {}
+  : HasIdentifierAndColour { new_good_definition },
+  	good_definition { new_good_definition },
+	price { new_good_definition.get_base_price() },
+	is_available { new_good_definition.get_is_available_from_start() },
+	supply_running_total { fixed_point_t::_0() },
+	total_supply_yesterday { fixed_point_t::_0() },
+	market_sell_order_quantities {},
+	market_sell_order_callbacks {}
+	{}
+
+void GoodInstance::add_market_sell_order(const fixed_point_t quantity, const std::function<void(SellResult)> callback) {
```

Comment:
Needs to be thread safe as it could be called from parallel code (processing pops, factories, rgos & governments in parallel for example).